presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -198,6 +200,64 @@ public static Slice stCentroid(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return serialize(createFromEsriGeometry(centroid, geometry.getEsriSpatialReference()));
     }
 
+    @Description("Returns the minimum convex geometry that encloses all input geometries")
+    @ScalarFunction("ST_ConvexHull")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice stConvexHull(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        if (!isValidGeometryForConvexHull(geometry)) {
+            return serialize(geometry);
+        }
+        OGCGeometry convexHull = geometry.convexHull();
+        long pointCount = getOGCGeometryPointCount(convexHull);
+        if (pointCount == 0) {
```

Comment:
@mbasmanova  I think this would happen when everything was in a single point.
You can see all those examples in the test.

Reply:
@tpeng-fb Got it. Generally speaking all the geospatial functions assume that input geometries are simple and valid (see the first paragraph in the docs below). Hence, we don't need to worry about input geometries that are *all single point* except when these are *exactly one point*.  E.g. we don't need to have `convexHull` function work on geometries like `MULTILINESTRING ((1 1), (1 1), (1 1, 1 1))` and `MULTIPOINT (1 1, 1 1, 1 1)`.

https://prestodb.io/docs/current/functions/geospatial.html

Reply:
@mbasmanova  I see, I can just revise it back

Reply:
@mbasmanova Just want to double check if we need to care about something like MUTIPOINT (1 1) or POLYGON (1 1)

Reply:
`MUTIPOINT (1 1)` and `GEOMETRY_COLLECTION (POINT (1 1))` are valid geometries. `POLYGON (1 1)` is not.

Reply:
@mbasmanova  In this situation I think this private method is still valid. But I think we can skip the checking for COLLECTION in case we take the option two you have mentioned above.